packages/flutter_tools/bin/xcode_backend.sh

```diff
@@ -251,7 +256,7 @@ LipoExecutable() {
 
   # Generate a merged binary from the architecture-specific executables.
   # Skip this step for non-fat executables.
-  if [[ ${#all_executables[@]} > 0 ]]; then
+  if [[ ${#all_executables[@]} -gt 0 ]]; then
```

Comment:
Oh interesting -- I thought these were equivalent when used inside double-square-bracket checks but not single-square-bracket checks (at least in ksh, zsh and bash). Is that not the case, or is this just for consistency? Agreed it makes sense for consistency either way.

Reply:
I believe this does a string comparison right. Maybe @jmagman has more insight here

Reply:
https://www.gnu.org/software/bash/manual/bash.html#Bash-Conditional-Expressions says
> When used with [[, the ‘<’ and ‘>’ operators sort lexicographically using the current locale. The test command uses ASCII ordering.

But I do think it's working now as-is since "1" is > "0", and there are well less than 10 archs.